app/src/main/java/org/p2p/wallet/infrastructure/network/NetworkModule.kt

```diff
@@ -103,4 +149,43 @@ object NetworkModule : InjectionModule {
             .addNetworkInterceptor(ContentTypeInterceptor())
             .build()
     }
+
+    private fun createCertificate(trustedCertificateIS: InputStream): SSLContext? {
+        val certificateFactory: CertificateFactory = CertificateFactory.getInstance("X.509")
+        val certificate: Certificate = trustedCertificateIS.use { trustedCertificateIS ->
+            certificateFactory.generateCertificate(trustedCertificateIS)
+        }
+
+        // creating a KeyStore containing our trusted CAs
+        val keyStoreType: String = KeyStore.getDefaultType()
+        val keyStore: KeyStore = KeyStore.getInstance(keyStoreType)
+        keyStore.load(null, null)
+        keyStore.setCertificateEntry("ca", certificate)
+
+        // creating a TrustManager that trusts the CAs in our KeyStore
+        val trustAlgorithm: String = TrustManagerFactory.getDefaultAlgorithm()
+        val trustManagerFactory: TrustManagerFactory = TrustManagerFactory.getInstance(trustAlgorithm)
+        trustManagerFactory.init(keyStore)
+
+        // creating an SSLSocketFactory that uses our TrustManager
+        val sslContext: SSLContext = SSLContext.getInstance("TLS")
```

Comment:
sonarlint warns (kotlin:S4423):

This rule raises an issue when an insecure TLS protocol version (i.e. a protocol different from "TLSv1.2", "TLSv1.3", "DTLSv1.2", or "DTLSv1.3") is used or allowed.
It is recommended to enforce TLS 1.2 as the minimum protocol version and to disallow older versions like TLS 1.0. Failure to do so could open the door to downgrade attacks: a malicious actor who is able to intercept the connection could modify the requested protocol version and downgrade it to a less secure version.

solution:
```
javax.net.ssl.SSLContext library:
val sc: SSLContext = SSLContext.getInstance("TLSv1.1") // Noncompliant
okhttp library:
val spec: ConnectionSpec = ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
                .tlsVersions(TlsVersion.TLS_1_1)  // Noncompliant
                .build()
```